api/include/opentelemetry/logs/logger_provider.h

```diff
@@ -2,6 +2,7 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #pragma once
+
 #ifdef ENABLE_LOGS_PREVIEW
 
 #  include "opentelemetry/common/key_value_iterable.h"
```

Comment:
Should we export `LoggerProvider` ? It has static variables also.

Reply:
Thanks for the catch, `LoggerProvider` should be exported. This PR makes trace and logs working, metrics will be added later based on the mechanism in this PR.